From 525bf2d377c21b41971a72d694a507bf2af0ada6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@kde.org>
Date: Tue, 16 Feb 2016 17:59:32 +0100
Subject: [PATCH] Fix IconItem not loading non-theme icons with name

Commit 5184ac introduced a small regression that caused IconItem
to no longer load non-theme icons with name set, for instance icons
coming from sni-qt, causing sni-qt apps to not show in systray.

BUG: 359388
FIXED-IN: 5.20
REVIEW: 127091
---
 src/declarativeimports/core/iconitem.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/declarativeimports/core/iconitem.cpp b/src/declarativeimports/core/iconitem.cpp
index 1d7921a..085f284 100644
--- a/src/declarativeimports/core/iconitem.cpp
+++ b/src/declarativeimports/core/iconitem.cpp
@@ -148,6 +148,10 @@ void IconItem::setSource(const QVariant &source)
                 //fail, use QIcon
                 } else {
                     m_icon = QIcon::fromTheme(sourceString);
+                    if (m_icon.isNull()) {
+                        // fallback for non-theme icons
+                        m_icon = source.value<QIcon>();
+                    }
                     delete m_svgIcon;
                     m_svgIcon = 0;
                     m_imageIcon = QImage();
-- 
2.4.10

